.github/actions/kibana-docker-image/action.yml

```diff
@@ -0,0 +1,87 @@
+name: Kibana Docker Image
+description: |
+  An Action to build and push Kibana docker images given a git ref.
+inputs:
+  git-repository:
+    description: The git repository to build the image from.
+    default: "elastic/kibana"
+    required: false
+  git-ref:
+    description: "The git ref of elastic/kibana to build the image from. (Default: default branch)"
+    required: false
+  serverless:
+    description: Whether to build the serverless image or not.
+    required: false
+    default: "false"
+  docker-registry:
+    description: Docker registry to publish the image to.
+    required: false
+    default: "docker.elastic.co"
+  docker-namespace:
+    description: Docker namespace to publish the image to.
+    required: false
+    default: "observability-ci"
+
+outputs:
+  ref:
+    description: The reference of the Docker image that was built in the format image:tag.
+    value: ${{ steps.vars.outputs.docker-reference }}
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.git-repository }}
+        ref: ${{ inputs.git-ref }}
+        path: kibana-repo-checkout
```

Comment:
**nit**: what do you think using the toplevel env: CHECKOUT_DIR: "kibana-repo-checkout"

then


```suggestion
        path: ${{ env.CHECKOUT_DIR }}
```

and in the other places too?

Reply:
https://github.com/elastic/apm-pipeline-library/pull/2348/commits/3deb3b43a6d5e21adb6724815fe16281f6a3cc65